### PR TITLE
Detect Error case in the xcatd and avoid loop in the updateflag.awk

### PR DESCRIPTION
**Pull request by cxhong**
Check description of issue #712
1) add trace message if didn't get hostname from gethostbyaddr() function in the xcatd
2) add trace message if didn't find this node in the nodels in the xcatd
3) avoid loop in the updateflag.awk

### CODE REVIEW

File: xCAT-server/sbin/xcatd [hunk 1]
```diff
@@ -377,6 +377,9 @@ sub do_installm_service {
         } else {
             ($client_name,$client_aliases) = gethostbyaddr($conn->peeraddr,AF_INET);
         }
+        unless ($client_name) {
+            xCAT::MsgUtils->message("S","xcatd unable to gethostbyaddr");
```
Comment (immarvin) [inline xCAT-server/sbin/xcatd]: suggest the information to be "xcatd received a connection request from unknown host with ip address \"$conn->peeraddr\", please check whether the reverse name resolution for it works correctly"

File: xCAT-server/sbin/xcatd [hunk 2]
```diff
@@ -403,7 +406,11 @@ sub do_installm_service {
             if ($node) {  # Means the source isn't valid
                 $validclient=1;
                 last;
+            } else {
+                xCAT::MsgUtils->message("S","Invalid client $client, please check nodels and it's name resolution");
```
Comment (immarvin) [inline xCAT-server/sbin/xcatd]: the message should better be "xcatd received a connection request from $client, which can not be found in xCAT nodelist table. The connection request will be ignored"

File: xCAT/postscripts/updateflag.awk [hunk 3]
```diff
@@ -6,27 +6,31 @@ BEGIN {
...
-		if($0 == "ready")
-			print flag |& ns
-		if($0 == "done")
-			break
-	}
+        if($0 == "ready")
+            print flag |& ns
+        if($0 == "done")
+            break
+        if(loop > 10)
```
Comment (immarvin) [inline xCAT/postscripts/updateflag.awk]: there should better be a message here, something like
````
             print "flag update failed" | "logger -t xcat -p local4.info"
````

**Comment by cxhong** [comment]
Thanks, @immarvin .

I modified the messages

**Comment by immarvin** [comment]
looks good to me, agree to merge

**Pull request merged**